identity/app/controllers/EditProfileController.scala

```diff
@@ -52,14 +54,31 @@ class EditProfileController(
   def submitAccountForm(): Action[AnyContent] = submitForm(AccountEditProfilePage)
   def submitPrivacyForm(): Action[AnyContent] = submitForm(PrivacyEditProfilePage)
 
+  def saveEmailPreferences: Action[AnyContent] =
+    csrfCheck {
+      authActionWithUser.async { implicit request =>
+        emailService.savePreferences().map { form  =>
+          if (form.hasErrors) {
+            val errorsAsJson = Json.toJson(
+              form.errors.groupBy(_.key).map { case (key, errors) =>
+                val nonEmptyKey = if (key.isEmpty) "global" else key
+                (nonEmptyKey, errors.map(e => play.api.i18n.Messages(e.message, e.args: _*)))
+              }
+            )
+            Forbidden(errorsAsJson)
```

Comment:
Is forbidden the right status here? Maybe bad request?

Reply:
Agreed. Refactoring in separate PR.